operators/multitenancy-operator/MultitenancyService.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const _ = require('lodash');
+const assert = require('assert');
+const Promise = require('bluebird');
+const logger = require('../../common/logger');
+const errors = require('../../common/errors');
+const CONST = require('../../common/constants');
+const catalog = require('../../common/models').catalog;
+const NotFound = errors.NotFound;
+const Gone = errors.Gone;
+const BaseService = require('../BaseService');
+const cf = require('../../data-access-layer/cf');
+const bosh = require('../../data-access-layer/bosh');
+const MultitenancyAgent = require('./MultitenancyAgent');
+const eventmesh = require('../../data-access-layer/eventmesh');
+
+class MultitenancyService extends BaseService {
+  constructor(guid, spaceId, plan, parameters, resourceType) {
+    super(plan);
+    this.guid = guid;
+    this.spaceId = spaceId;
+    this.parameters = parameters;
+    this.director = bosh.director;
+    this.cloudController = cf.cloudController;
+    this.resourceType = resourceType;
+    this.agent = new MultitenancyAgent(this.settings.agent);
```

Comment:
It would be good if we refrain from instantiating objects directly. Agent can be passed so that service class code can be reused.

Reply:
Thanks for suggesting !
Tried doing but not getting some settings outside this constructor. So, created object inside constructor only.  I will further analyze and take this change later.